Modules/Module_Processing/Module_ASL/xASL_wrp_ProcessM0.m

```diff
@@ -120,10 +131,40 @@ function xASL_wrp_ProcessM0(x)
 
 			% Find the smallest of the previously selected positive nonzero TEs
 			minTE = min(uniqueNonzeroTE);
-			fprintf('Multi-TE M0 present. Using the shortest TE=%.1f ms for M0 calibration.\n', minTE*1000);
+			% JSON values were converted from BIDS (seconds) to Legacy (millisecond) during JSON reading
+			fprintf('Multi-TE M0 present. Using the shortest TE = %.1f ms for M0 calibration.\n', minTE);
+
+			% Find the correct indices and update the jsonM0 parameter
+			indMinTE = jsonM0.Q.EchoTime == minTE;
+			jsonM0.Q.EchoTime = jsonM0.Q.EchoTime(indMinTE);
+			if isfield(jsonM0, 'RepetitionTime') && length(jsonM0.RepetitionTime) > 1
+				jsonM0.RepetitionTime = jsonM0.RepetitionTime(indMinTE);
+			end
+
+			% Select the minimal TEs
+			imM0 = imM0(:, :, :, indMinTE);
+		end
+	end
 
-			% Average M0 image across all these minimal TEs
-			imM0 = xASL_stat_MeanNan(imM0(:,:,:,jsonM0.Q.EchoTime == minTE), 4);
+%% 0D. Use the M0 with the TR closest to 2000 ms (if we have multiple M0s)
+	if isfield(jsonM0, 'RepetitionTime')
+		uniqueNonzeroTR = unique(jsonM0.RepetitionTime(jsonM0.RepetitionTime > 0));
+
+		% Check if we have multiple TR to decide to select a single value only
+		if length(uniqueNonzeroTR) > 1
+			% We currently do not do inversion fitting in multi-TR M0, so we issue a warning and use the TR value closest to 2000ms

```

Comment:
This comment also belongs to us not fitting multi-TE M0 currently right? Or do we now try to fit it?

Reply:
As explained above. Makes no sense to fit multi-TE for M0 quantification. So that's why I don't even mention it. I've added a comment about this.